crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -230,14 +231,14 @@ pub struct SyncRequestListFilters {
     /// returned regardless of type. This can be used to get the initial set of spaces for an
     /// account.
     #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub room_types: Vec<String>,
+    pub room_types: Vec<Option<RoomType>>,
```

Comment:
This change seems to make sense, but it would be great if we can have a review of someone familiar with sliding-sync, @Hywan maybe?

It looks like the docs need to be updated as they still talk about strings.

Reply:
Why would it be `Vec<Option<_>>` though? That would mean there can be `null`s inside the array.

Reply:
There can be, normal rooms don't have a `type`, so they are represented by `null`.

Reply:
We could have a `None` variant in `RoomType` 😅 
(though then `as_str` would have to return an `Option<&str>`)

Reply:
It looks like we already have [`RoomTypeFilter`](https://docs.ruma.dev/ruma/directory/enum.RoomTypeFilter.html).

Reply:
would it make sense to add a From impl for RoomTypeFilter that takes an Option<RoomType> then? Or is there already an easy way to do that :p

Reply:
It is already possible to do that, as explained in the docs of the type.

Reply:
oh sorry, i meant an Option<RoomType>

Reply:
Ah, yeah I guess we can add one.

Reply:
It is true that [the MSC3575 specifies it's an array of room types](https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md#room-list-parameters). However, we are working a simplified sliding sync version so does it make sense to modify this codebase (in the simplified version, these two filters are removed)? I'm OK to merge this though, as it's more correct.

Reply:
I know most of the filters are going away, but i thought `not_room_types` was staying?

And it definitely made things comfier when implementing this on the server side, and will also make it easier when moving that code to support simplified sliding sync

Reply:
I'm alright with this modification. And yes, `is_invite` and `not_room_types` are kept. I was confused.